plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/sampler/ProbabilisticSampler.java

```diff
@@ -67,7 +74,14 @@ public SamplingResult shouldSample(
         Attributes attributes,
         List<LinkData> parentLinks
     ) {
-        return getSampler().shouldSample(parentContext, traceId, name, spanKind, attributes, parentLinks);
+        double newSamplingRatio = telemetrySettings.getSamplingProbability();
+        if (isSamplingRatioChanged(newSamplingRatio)) {
+            synchronized (this) {
+                this.samplingRatio = newSamplingRatio;
+                defaultSampler = Sampler.traceIdRatioBased(samplingRatio);
+            }
+        }
+        return defaultSampler.shouldSample(parentContext, traceId, name, spanKind, attributes, parentLinks);
```

Comment:
The `fallbackSampler` should be used somehow, but it is not:
```suggestion
        SamplingResult  result = defaultSampler.shouldSample(parentContext, traceId, name, spanKind, attributes, parentLinks);
        if (result.getDecision() != DROP) {
            return fallbackSampler.shouldSample(parentContext, traceId, name, spanKind, attributes, parentLinks);
        } else {
            return result;
        }
```

Reply:
+1